pkg/apis/networking/validation/validation.go

```diff
@@ -277,26 +282,24 @@ func validateHTTPIngressRuleValue(httpIngressRuleValue *networking.HTTPIngressRu
 	if len(httpIngressRuleValue.Paths) == 0 {
 		allErrs = append(allErrs, field.Required(fldPath.Child("paths"), ""))
 	}
-	for i, rule := range httpIngressRuleValue.Paths {
-		if len(rule.Path) > 0 {
-			if !strings.HasPrefix(rule.Path, "/") {
-				allErrs = append(allErrs, field.Invalid(fldPath.Child("paths").Index(i).Child("path"), rule.Path, "must be an absolute path"))
+	for i, path := range httpIngressRuleValue.Paths {
+		idxPath := fldPath.Child("paths").Index(i).Child("path")
+		if !strings.HasPrefix(path.Path, "/") {
+			allErrs = append(allErrs, field.Invalid(idxPath, path.Path, "must be an absolute path"))
+		}
+
+		if path.PathType != nil {
+			if !supportedPathTypes.Has(string(*path.PathType)) {
+				allErrs = append(allErrs, field.NotSupported(fldPath.Child("paths").Index(i).Child("pathType"), path.PathType, supportedPathTypes.List()))
 			}
-			// TODO: More draconian path regex validation.
-			// Path must be a valid regex. This is the basic requirement.
-			// In addition to this any characters not allowed in a path per
-			// RFC 3986 section-3.3 cannot appear as a literal in the regex.
-			// Consider the example: http://host/valid?#bar, everything after
-			// the last '/' is a valid regex that matches valid#bar, which
-			// isn't a valid path, because the path terminates at the first ?
-			// or #. A more sophisticated form of validation would detect that
-			// the user is confusing url regexes with path regexes.
-			_, err := regexp.CompilePOSIX(rule.Path)
-			if err != nil {
-				allErrs = append(allErrs, field.Invalid(fldPath.Child("paths").Index(i).Child("path"), rule.Path, "must be a valid regex"))
+
+			if *path.PathType == networking.PathTypeExact || *path.PathType == networking.PathTypePrefix {
```

Comment:
Is there more we can or should validate here?

Reply:
Potentially yes. The KEP does not specify any additional validation, but there could be value in at least validating that the characters provided for these path types are URL safe. Happy to add that here and in the KEP if you think it would be valuable. It doesn't seem like there is any universally safe limit we could apply to the length here, but open to ideas on what else would make sense to validate here.